pkg/services/automation/automation_test.go

```diff
@@ -0,0 +1,354 @@
+package automation
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	ctx context.Context
+
+	emptyRepoURL = gitproviders.RepoURL{}
+)
+
+func createRepoURL(url string) gitproviders.RepoURL {
+	repoURL, err := gitproviders.NewRepoURL(url)
+	Expect(err).NotTo(HaveOccurred())
+
+	return repoURL
+}
+
+var _ = Describe("Generate manifests", func() {
+	var app models.Application
+
+	var _ = BeforeEach(func() {
+		app = models.Application{
+			Name:                "bar",
+			Namespace:           wego.DefaultNamespace,
+			HelmSourceURL:       "",
+			GitSourceURL:        createRepoURL("ssh://git@github.com/foo/bar.git"),
+			Branch:              "main",
+			Path:                "./kustomize",
+			AutomationType:      models.AutomationTypeKustomize,
+			SourceType:          models.SourceTypeGit,
+			HelmTargetNamespace: "",
+		}
+
+		gitProviders.GetDefaultBranchReturns("main", nil)
+
+		ctx = context.Background()
+	})
+
+	Context("add app with config in app repo", func() {
+		BeforeEach(func() {
+			app.GitSourceURL = createRepoURL("ssh://git@github.com/foo/bar.git")
+			app.ConfigURL = emptyRepoURL
+		})
+
+		Describe("generates source manifest", func() {
+			It("creates GitRepository when source type is git", func() {
+				app.SourceType = models.SourceTypeGit
+				_, err := automationSvc.GenerateAutomation(ctx, app, "test-cluster")
```

Comment:
That's fair, but just FYI... I didn't really write those tests. Those are all the same tests we had before except that they take a `model.Application` and invoke `automation.GenerateAutomation` rather than `AddParams` and `app.Add`. Example:

```
		Describe("generates source manifest", func() {
			It("creates GitRepository when source type is git", func() {
				addParams.SourceType = wego.SourceTypeGit
				err := appSrv.Add(addParams)
				Expect(err).ShouldNot(HaveOccurred())

				Expect(fluxClient.CreateSourceGitCallCount()).To(Equal(1))

				name, url, branch, secretRef, namespace := fluxClient.CreateSourceGitArgsForCall(0)
				Expect(name).To(Equal("bar"))
				Expect(url.String()).To(Equal("ssh://git@gitlab.com/group/subgroup/bar.git"))
				Expect(branch).To(Equal("main"))
				Expect(secretRef).To(Equal("wego-test-cluster-bar"))
				Expect(namespace).To(Equal(wego.DefaultNamespace))
			})
		})
```

I figured just migrating them would be okay given the huge set of changes that are already in here. WDYT about redoing the tests in a separate PR w/ the `RepoWriter` tests? If that causes too much heartburn, I can do it now.

Reply:
@jrryjcksn Ok them I am cool with migrating them for now and "fixing" them later :+1:

Reply:
I've been looking at this some more... Right now we're checking that we pass the right args to the `flux` commands for GitRepositories, HelmRepositories, Kustomizations, and HelmReleases. If we check the return value from the generation, we'll be fragile in the face of changes in the manifests that flux generates. Is that what we want to check here? I can make sure that the things we generate ourselves (app.yaml, app kustomize, and user kustomize) are correct but maybe we shouldn't be explicitly checking the output of flux here for the things _it_ generates :thinking: